datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -1487,10 +1492,12 @@ impl HashJoinStream {
         &mut self,
         cx: &mut std::task::Context<'_>,
     ) -> Poll<Option<ArrowResult<RecordBatch>>> {
+        let build_timer = self.join_metrics.build_time.timer();
```

Comment:
This will be timing the overall clock time (not the cpu time) of the build. As long as that is what you are trying to time 👍